scipy/sparse/linalg/_propack/setup.py

```diff
@@ -0,0 +1,68 @@
+from os.path import join
+import pathlib
+
+
+def check_propack_submodule():
+    if not (pathlib.Path(__file__).parent / 'PROPACK/README').exists():
+        raise RuntimeError("Missing the `PROPACK` submodule! Run "
+                           "`git submodule update --init` to fix this.")
+
+
+def configuration(parent_package='', top_path=None):
+    from numpy.distutils.system_info import get_info, NotFoundError
+    from numpy.distutils.misc_util import Configuration
+    config = Configuration('_propack', parent_package, top_path)
+    lapack_opt = get_info('lapack_opt')
+    if not lapack_opt:
+        raise NotFoundError('no lapack/blas resources found')
+
+    #------------------------------------------------------------
+    # Set up the libraries.
+    #  We need a different python extension file for each, because
+    #  names resue between functions in the LAPACK extensions.  This
+    #  could probably be remedied with some work.
+    type_dict = dict(s='single',
+                     d='double',
+                     c='complex8',
+                     z='complex16')
+    check_propack_submodule()
+    base = pathlib.Path(__file__).parent / 'PROPACK'
+    for prefix, directory in type_dict.items():
+        lapack_lib = f'sp_{prefix}lapack_util'
+        propack_lib = f'_{prefix}propack'
+
+        # Preprocess sources to remove $OMP references
```

Comment:
@rgommers It's not my favorite thing I've ever done, but this seems to get the job done (i.e. builds and test pass without undefined symbol errors).  The strategy is to rewrite files with `$OMP` and `_OPENMP` and substitute those substrings with something else that the compiler/linker will ignore.  I did this because I found adding `undef_macros` to `config.add_library` a bigger pain than I had anticipated and this seemed and I didn't want to bother with the cross-platform checking.  Happy to try out the `pre_build_hook` if you'd like to go down that route

Reply:
This seems fine to me, the pre-build hook isn't pretty either. 

It would be nice to have this as a separate script instead of in `setup.py`. Then we can use it from Meson as well. And add `all_newer` as in `scipy/special/_generate_pyx.py` so that the rewriting is only done once rather than on every rebuild.

Reply:
I was ultimately unhappy with this solution because
- I don't want to modify the contents of a submodule in place
- I couldn't figure out how to write/read modified sources in the build directory (might be a way, I just couldn't figure it out -- probably easy to do with Meson)
- I didn't want to write another modified PROPACK directory in the source

But, by scouring the CPython `distutils` implementation, I did figure out how to give `config.add_library()` a set of macros to undefine!  So I applied a few more patches to the PROPACK repo to allow a build without any OpenMP enabled by adding `#ifdef` guards around all `$OMP` calls and now everything seems to be working much more cleanly

Thanks for helping track this down, I probably wouldn't have been able to pin it down what conda was doing

Reply:
Makes sense. Nice workaround to solve this!

> probably easy to do with Meson, I know this is trivial with cmake)

yep, it just passes you the build dir. it's not only easy, it's mandatory - no in-place modifications are allowed, everything must be out of tree.